reason-react-native/src/apis/Style.md

```diff
@@ -4,4 +4,1297 @@ title: Style
 officialDoc: https://facebook.github.io/react-native/docs/style
 ---
 
-@todo
+Styles accepted by most components (thanks to a `style` prop) are defined using
+style functions that prepare required object for React Native. The style names
+and values usually match how CSS works on the web, except names are written
+using camel casing, e.g `backgroundColor` rather than `background-color`.
+
+The `style` prop accepts values returned by the `style()` constructor. That's
+the simplest way to pass styles. You can also pass an array or list of styles -
+the last style in the array has precedence, so you can use this to mix & inherit
+styles.
+
+⚠️ _Note that when a component grows in complexity, it is often cleaner to use
+[`StyleSheet.create`](/bs-react-native/en/docs/apis/StyleSheet/) to define
+several styles in one place_.
+
+We have made different style constructors because React Native have various
+components that accept different styles props. For example `View` doesn't accept
+`color` (only `Text` does).
+
+## Table of Content
+
+- [Style Example](#style-example)
+- [Style Units](#style-units)
+- [Style Function](#style-functions)
+  - [`Style.style`](#style-style)
+  - [`Style.viewStyle`](#style-viewStyle)
+  - [`Style.textStyle`](#style-textStyle)
+  - [`Style.imageStyle`](#style-imageStyle)
+- [Style Props](#style-props)
+  - [Layout Style Props](#layout-style-props)
+  - [Shadow Style Props](#shadow-style-props)
+  - [Transform Style Props](#transform-style-props)
+  - [View Style Props](#view-style-props)
+  - [Text Style Props](#text-style-props)
+  - [Image Style Props](#image-style-props)
+  - [Unsafe Style Props](#unsafe-style-props)
+- [Style Helpers](#style-helpers)
+  - [`Style.array`](#style-array`)
+  - [`Style.arrayOption`](#style-arrayoption`)
+  - [`Style.list`](#style-list`)
+  - [`Style.listOption`](#style-listoption`)
+
+## Style Example
+
+Since an example is worth a thousand words...
+
+```reason
+open ReactNative;
+
+let styles =
+  Style.(
+    StyleSheet.create({
+      "container":
+        viewStyle(
+          ~maxHeight=600.->pt,
+          ~width=800.->pct,
+          ~justifyContent=`flexStart,
+          ~alignItems=`center,
+          ~margin=auto,
+          (),
+        ),
+      "cornerThing":
+        viewStyle(
+          ~position=`absolute,
+          ~top=100.->pt,
+          ~right=(-20.)->pt,
+          ~transform=[|rotate(~rotate=4.->deg)|],
+          (),
+        ),
+      "text": textStyle(~textTransform=`uppercase, ()),
+    })
+  );
+
+[@react.component]
+let make = (~isSomething) => {
+  <View style=styles##container>
+    <View style=styles##cornerThing>
+      <Text
+        style=Style.(
+          arrayOption([|
+            Some(styles##text),
+            isSomething ? Some(style(~opacity=0.05, ())) : None,
+          |])
+        )
+      />
+    </View>
+  </View>;
+};
+```
+
+## Style Units
+
+### `size`
+
+`size` is used in various place to express logical pixels (`pt` function),
```

Comment:
how about this rephrasing:
`size` is required by various style props, expressed as logical pixels (`pt` function),
percentage (`pct` function) & also `auto` (inline string, edge case for
`margin`).

Reply:
slight improvement

Reply:
no worries :)